platforms/software/software-diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/insight/DependencyInsightReporter.java

```diff
@@ -172,6 +179,23 @@ private static void collectErrorMessages(Throwable failure, TreeFormatter format
         }
     }
 
+    /**
+     * Multi-cause exceptions by default aggregate the resolutions of their causes.
+     * If a cause has resolutions, skip emitting the parent exception's resolutions,
+     * so we can correctly associate the resolution with the child.
+     */
+    private static boolean shouldSkipResolutions(Throwable throwable) {
```

Comment:
We should instead print the set of parent resolutions minus the set of all child resolutions, in order to handle cases where the multi cause implementation does not simply aggregate all of its children